File: client/src/model/game/card/PokerCard.ts
import {Card} from "./Card";

export enum Suit {
  HEARTS, DIAMONDS, CLUBS, SPADES, SPECIAL
}

export enum Rank {
  ACE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN,
  JACK, QUEEN, KING, JOKER, HIDDEN
}

export class PokerCard implements Card {
  private _suit: Suit;
  private _rank: Rank;

  constructor(suit: Suit, rank: Rank) {
    this._rank = rank;
    this._suit = suit;
  }

  get suit(): Suit {
    return this._suit;
  }

  get rank(): Rank {
    return this._rank;
  }

  imageUrl() {
    if (this._rank == Rank.JOKER) return "./cards/Joker.svg";
    if (this._rank == Rank.HIDDEN) return "./cards/Back.svg";
    let value;
    let suit;
    switch (this._rank) {
      case Rank.ACE:
        value = 'A';
        break;
      case Rank.TWO:
        value = '2';
        break;
      case Rank.THREE:
        value = '3';
        break;
      case Rank.FOUR:
        value = '4';
        break;
      case Rank.FIVE:
        value = '5';
        break;
      case Rank.SIX:
        value = '6';
        break;
      case Rank.SEVEN:
        value = '7';
        break;
      case Rank.EIGHT:
        value = '8';
        break;
      case Rank.NINE:
        value = '9';
        break;
      case Rank.TEN:
        value = 'T';
        break;
      case Rank.JACK:
        value = 'J';
        break;
      case Rank.QUEEN:
        value = 'Q';
        break;
      case Rank.KING:
        value = 'K';
        break;
    }
    switch (this._suit) {
      case Suit.SPADES:
        suit = 'S';
        break;
      case Suit.HEARTS:
        suit = 'H';
        break;
      case Suit.CLUBS:
        suit = 'C';
        break;
      case Suit.DIAMONDS:
        suit = 'D';
        break;
    }
    return `./cards/${value}${suit}.svg`;
  }

  keyString(): string {
    return `${Rank[this._rank]} of ${Suit[this._suit]}`;
  }
}

File: server/src/main/java/dev/mdaven/familygames/game/card/PokerCard.java
package dev.mdaven.familygames.game.card;

import java.util.Objects;

public class PokerCard {

    public enum Suit {
        CLUBS, DIAMONDS, HEARTS, SPADES, SPECIAL
    }

    public enum Rank {
        ACE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN, JACK, QUEEN, KING, JOKER;
    }

    private final Rank rank;
    private final Suit suit;

    public PokerCard(Rank rank, Suit suit) {
        this.rank = rank;
        this.suit = suit;
    }

    public Rank getRank() {
        return rank;
    }

    public Suit getSuit() {
        return suit;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        PokerCard card = (PokerCard) o;
        return rank == card.rank && suit == card.suit;
    }

    @Override
    public int hashCode() {
        int result = Objects.hashCode(rank);
        result = 31 * result + Objects.hashCode(suit);
        return result;
    }

    @Override
    public String toString() {
        if(suit == Suit.SPECIAL) return rank.name().toLowerCase();
        return rank.name().toLowerCase() + " of " + suit.name().toLowerCase();
    }
}
